registry/auth.go

```diff
@@ -56,7 +62,13 @@ func NewRequestAuthorization(authConfig *AuthConfig, registryEndpoint *Endpoint,
 }
 
 func (auth *RequestAuthorization) getToken() (string, error) {
-	// TODO check if already has token and before expiration
+	auth.tokenLock.Lock()
+	defer auth.tokenLock.Unlock()
+	if auth.tokenExpiration.After(time.Now()) {
```

Comment:
Nit pick: may want to this to read the other way:

```
if time.Now().Before(auth.tokenExpiration) {
...
}
```

You may also want to call `time.Now` only once for the function to avoid skew (which really doesn't matter but it's good practice).